src/Pulsar.Client/Api/Producer.fs

```diff
@@ -21,6 +21,7 @@ type ProducerState = {
 type Producer private (producerConfig: ProducerConfiguration, lookup: BinaryLookupService) =
     let producerId = Generators.getNextProducerId()
     let messages = ConcurrentDictionary<SequenceId, TaskCompletionSource<MessageId>>()
+    let pendingMessages = ConcurrentQueue<ProducerMessage>()
```

Comment:
All messages are processed through the mailbox, so regular Queue looks to be enough here